src/fastmeteo/data/arpege.py

```diff
@@ -0,0 +1,113 @@
+import tempfile
+from datetime import datetime, timedelta, timezone
+from io import BytesIO
+from pathlib import Path
+from typing import Any
+
+import httpx
+import numpy as np
+import pandas as pd
+import xarray as xr
+from tqdm.auto import tqdm
+
+from .. import aero
+from ..core.grid import Grid
+
+tempdir = Path(tempfile.gettempdir())
+
+bare_url = "https://object.data.gouv.fr/meteofrance-pnt/pnt/"
+
+
+def download_with_progress(url: str) -> BytesIO:
+    with httpx.stream("GET", url) as r:
+        total_size = int(r.headers.get("Content-Length", 0))
+        buffer = BytesIO()
+        with tqdm(
+            total=total_size, unit="B", unit_scale=True, desc=url.split("/")[-1]
+        ) as progress_bar:
+            for chunk in r.iter_bytes():
+                buffer.write(chunk)
+                progress_bar.update(len(chunk))
+        buffer.seek(0)
+        return buffer
+
+
+class Arpege(Grid):
```

Comment:
The full data source is described here:
https://www.data.gouv.fr/fr/datasets/donnees-pnt-retention-14-jours/